sklearn/metrics/_ranking.py

```diff
@@ -754,7 +754,7 @@ def _binary_clf_curve(y_true, y_score, pos_label=None, sample_weight=None):
     pos_label = _check_pos_label_consistency(pos_label, y_true)
 
     # make y_true a boolean vector
-    y_true = y_true == pos_label
+    y_true = y_true.astype(bool)
```

Comment:
Thank you for responding. I have put together a toy example. Could you please explain the behavior?
[bug_example.txt](https://github.com/scikit-learn/scikit-learn/files/8100759/bug_example.txt)

Reply:
This is not the best place to ask ML questions since PRs are used for development. https://github.com/scikit-learn/scikit-learn/discussions or stackoverflow are better usage questions.

As a quick response, `pos_label=0` changes the interpretation of `y_true` to mean 0 is the positive class. `y_pred` was constructed to do well when `pos_label=1`, so it will do poorly when 0 is the positive class. (`y_pred` is basically ranking in the reverse direction)